File: src/documents/samples/adience/adience_example.cs
using UnityEngine;
using AdBoost;
using Soomla.Store;
using System.Collections.Generic;
using System.Linq;

public class MyBehaviour : MonoBehaviour {

    private void Start() {

        // Register SOOMLA Store event handlers
        StoreEvents.OnMarketPurchase += MyBehaviour.OnMarketPurchase;

        // Initialize SOOMLA Store
        // Assumes you've implemented your store assets
        SoomlaStore.Initialize(new YourStoreAssetsImplementation());
    }

    private static void OnMarketPurchase(PurchasableVirtualItem pvi, string payload, Dictionary<string, string> extra) {
        PurchaseWithMarket purchaseWithMarket = pvi.PurchaseType as PurchaseWithMarket;

        if (purchaseWithMarket != null) {
            MarketItem marketItem = purchaseWithMarket.MarketItem;
            if (Application.platform == RuntimePlatform.Android) {
                EventTracker tracker = AdBoostUnity.getEventTracker();
                tracker.sendRevenueEvent(
                    marketItem.MarketCurrencyCode,
                    marketItem.Price,
                    pvi.ItemId
                );
            }
        }
    }
}

File: src/files/samples/adience/adience_example.java
package com.mycompany.myapp;

import android.app.Activity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.adience.adboost.AdBoost;
import com.adience.sdk.EventTracker;
import com.soomla.BusProvider;
import com.soomla.Soomla;
import com.soomla.store.domain.MarketItem;
import com.soomla.store.domain.PurchasableVirtualItem;
import com.soomla.store.events.MarketPurchaseEvent;
import com.soomla.store.purchaseTypes.PurchaseWithMarket;
import com.squareup.otto.Subscribe;

import java.util.HashMap;


public class MainActivity extends Activity {

		@Override
		protected void onCreate(Bundle savedInstanceState) {
				super.onCreate(savedInstanceState);
				setContentView(R.layout.activity_main);
				AdBoost.appStarted();

				// Register this instance to receive SOOMLA events.
				BusProvider.getInstance().register(this);

				// Initialize SOOMLA Store
				// Assumes you've implemented your store assets.
				Soomla.initialize("<GAME SECRET GOES HERE>");
				SoomlaStore.getInstance().initialize(new YourStoreAssetsImplementation());
		}

		@Subscribe
		public void onMarketPurchase(MarketPurchaseEvent marketPurchaseEvent) {
				PurchasableVirtualItem pvi = marketPurchaseEvent.PurchasableVirtualItem;
				PurchaseWithMarket purchaseWithMarket =
						PurchaseWithMarket.class.isInstance(pvi.getPurchaseType()) ?
						(PurchaseWithMarket)pvi.getPurchaseType() : null;

				if(purchaseWithMarket != null) {
						MarketItem marketItem = purchaseWithMarket.getMarketItem();
						HashMap<String, String> extraInfo = marketPurchaseEvent.ExtraInfo;
						EventTracker tracker = AdBoost.getEventTracker();
						tracker.sendRevenueEvent(
								marketItem.getMarketCurrencyCode(),
								marketItem.getPrice(),
								pvi.getItemId()
						);
				}
		}
}
